src/test/java/com/microsoft/sqlserver/jdbc/unit/lobs/LobsStreamingTest.java

```diff
@@ -0,0 +1,282 @@
+        StringBuilder salt = new StringBuilder();
+        Random rnd = new Random();
+        while (salt.length() < length) {
+            int index = (int) (rnd.nextFloat() * validCharacters.length());
+            salt.append(validCharacters.charAt(index));
+        }
+        String saltStr = salt.toString();
+        return saltStr;
+
+    }
+
+    private String getStringFromInputStream(InputStream is) {
+        try (@SuppressWarnings("resource")
+        java.util.Scanner s = new java.util.Scanner(is, java.nio.charset.StandardCharsets.US_ASCII)
+                .useDelimiter("\\A")) {
+            return s.hasNext() ? s.next() : "";
+        }
+    }
+
+    private String getStringFromReader(Reader r, long l) throws IOException {
+        // read the Reader contents into a buffer and return the complete string
+        final StringBuilder stringBuilder = new StringBuilder((int) l);
+        char[] buffer = new char[(int) l];
+        while (true) {
+            int amountRead = r.read(buffer, 0, (int) l);
+            if (amountRead == -1) {
+                break;
+            }
+            stringBuilder.append(buffer, 0, amountRead);
+        }
+        return stringBuilder.toString();
+    }
+    
+    @Test
+    @DisplayName("testLengthAfterStream")
+    public void testLengthAfterStream() throws SQLException, IOException {
+        try (Connection conn = DriverManager.getConnection(connectionString); Statement stmt = conn.createStatement()) {
+            stmt.execute("CREATE TABLE [" + tableName + "] (id int, lobValue varchar(max))");
+            ArrayList<String> lobs = new ArrayList<>();
+            IntStream.range(0, LOB_ARRAY_SIZE).forEach(i -> lobs
+                    .add(getRandomString(ThreadLocalRandom.current().nextInt(LOB_LENGTH_MIN, LOB_LENGTH_MAX))));
+
+            try (PreparedStatement pstmt = conn.prepareStatement("INSERT INTO [" + tableName + "] VALUES(?,?)")) {
+                for (int i = 0; i < lobs.size(); i++) {
+                    Clob c = conn.createClob();
+                    c.setString(1, lobs.get(i));
+                    pstmt.setInt(1, i);
+                    pstmt.setClob(2, c);
+                    pstmt.addBatch();
+                }
+                pstmt.executeBatch();
+
+                ResultSet rs = stmt.executeQuery("SELECT * FROM [" + tableName + "] ORDER BY id ASC");
+                while (rs.next()) {
+                    Clob c = rs.getClob(2);
+                    Reader r = c.getCharacterStream();
+                    long clobLength = c.length();
+                    String recieved = getStringFromReader(r,clobLength);// streaming string
+                    assertEquals(lobs.get(rs.getInt(1)), recieved);// compare streamed string to initial string
+                }
+                rs.close();
+            }
+            stmt.execute("DROP TABLE [" + tableName + "]");

```

Comment:
Drop table in finally block to avoid table leftovers in case exception occurs.

Reply:
stmt was created in try-catch, cannot access in finally block.

Reply:
1. There is already `dropTableIfExists()` in Utils.java, consider using it.
2. Move the `drop table` into a `finally` block. You can either  create a new `Statement` for it or just use regular try-catch blocks.